### PR TITLE
feat(projects): add htop.dev package

### PR DESCRIPTION
**Pull request by mfts**
I have issues building this locally. 

```
configure: error: in `/Users/mfts/.tea/htop.dev/src/v3.2.1':
configure: error: C compiler cannot create executables
See `config.log' for more details
error: Uncaught (in promise) Error: cmd failed: 77: /Users/mfts/.tea/htop.dev/src/build.sh
```

### CODE REVIEW

File: projects/htop.dev/package.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+build:
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
+    gnu.org/autoconf: '*'
+    gnu.org/automake: '*'
+    invisible-island.net/ncurses: 6
+  script: |
+    ./autogen.sh && \
```
Comment (mxcl) [inline projects/htop.dev/package.yml]: they don’t provide a tarball with `./configure` already generated?

File: projects/htop.dev/package.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    ./configure --prefix=/.tea/htop.dev/ && \
```
Comment (mxcl) [inline projects/htop.dev/package.yml]: `--prefix={{prefix}}` this is the wrong prefix

File: projects/htop.dev/package.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+distributable:
+  # url: https://github.com/htop-dev/htop/archive/refs/tags/{{version}}.tar.gz
+  url: https://github.com/htop-dev/htop/releases/download/{{version}}/htop-{{version}}.tar.xz
+  strip-components: 1
+
+versions:
+  github: htop-dev/htop/tags
+  strip: /v/
```
Comment (mxcl) [inline projects/htop.dev/package.yml]: I believe stripping the `v` is automatic, but maybe not.

File: projects/htop.dev/package.yml (hunk 1 above)
Comment (mfts) [reply projects/htop.dev/package.yml]: In their ["compile from source"](https://github.com/htop-dev/htop#compile-from-source) it looks like `./configure` is provided.

File: projects/htop.dev/package.yml (hunk 2 above)
Comment (mfts) [reply projects/htop.dev/package.yml]: What is the correct prefix? Their docs (https://github.com/htop-dev/htop#install) say the `make install` will install it to `/usr/local` by default if I don't change the prefix in configure... I assumed I want it installed in my `.tea` directory.

File: projects/htop.dev/package.yml (hunk 1 above)
Comment (mxcl) [reply projects/htop.dev/package.yml]: Yeah so probs you don’t need to do the `autogen.sh`

File: projects/htop.dev/package.yml (hunk 2 above)
Comment (mxcl) [reply projects/htop.dev/package.yml]: the correct prefix is `{{prefix}}`. the mustaches are expanded to `TEA_PREFIX/PKG.com/v1.2.3`

You can see we do the same in all package.ymls

File: projects/htop.dev/package.yml (hunk 1 above)
Comment (mfts) [reply projects/htop.dev/package.yml]: my bad I misread your question. `configure.ac` is provided so we do need `autogen.sh`

Comment (mxcl) [reply projects/htop.dev/package.yml]: There's a `configure`, please remove the autogen and deps:
image

File: projects/htop.dev/package.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+build:
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
+    gnu.org/autoconf: '*'
+    gnu.org/automake: '*'
+    invisible-island.net/ncurses: 6
+  script: |
+    ./autogen.sh && \
+    ./configure --prefix={{prefix}} && \
```
Comment (mxcl) [inline projects/htop.dev/package.yml]: Don’t `&& \`.

We prefer to have distinct invocations of each line for easier to read logs.

File: projects/htop.dev/package.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+dependencies:
+  invisible-island.net/ncurses: 6
+
+build:
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
+    gnu.org/autoconf: '*'
+    gnu.org/automake: '*'
+    invisible-island.net/ncurses: 6
```
Comment (mxcl) [inline projects/htop.dev/package.yml]: no need to specify this in `build` deps since it is a runtime dep

**Pull request merged**